### PR TITLE
ARTEMIS-4780 - Artemis web console does not  set artemisJmxDomain

### PR DESCRIPTION
**Pull request by andytaylor**
expose a config file that can be shaded to change the domain used

### CODE REVIEW

File: artemis-console-extension/artemis-extension/src/artemis-extension/artemis/config-manager.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { log } from "./globals"
```
Comment (gemmellr) [inline artemis-console-extension/artemis-extension/src/artemis-extension/artemis/config-manager.ts:1]: Missing licence header failed the build on the RAT check. Related, the RAT check step only uploaded one result file, but they are generated per-module so the failure isnt actually in the uploaded file.

File: artemis-console-extension/artemis-extension/public/artemisconfig.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (gemmellr) [inline artemis-console-extension/artemis-extension/public/artemisconfig.json:1]: This one probably fails the RAT check too.

File: artemis-console-extension/artemis-extension/src/artemis-extension/artemis/config-manager.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      }
+    
+
+    private async loadConfig(): Promise<Artemisconfig> {
+    
+        try {
+            const res = await fetch(ARTEMISCONFIG_JSON)
+         if (!res.ok) {
+
+            log.info("Unable to load Artemis Config using Default");
```
Comment (gemmellr) [inline artemis-console-extension/artemis-extension/src/artemis-extension/artemis/config-manager.ts:39]: Indentation seems wonky.

Comma needed to make the log message more understandable: ", using Default"

File: artemis-console-extension/artemis-extension/src/artemis-extension/artemis/config-manager.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                    }
+                }
+            }
+
+            const config = await res.json()
+
+            log.info("Loaded Artemis Config:", config);
+            return config
+        } catch (err) {
+            log.info("Unable to load Artemis Config using Default");
```
Comment (gemmellr) [inline artemis-console-extension/artemis-extension/src/artemis-extension/artemis/config-manager.ts:52]: Comma needed to make the log message more understandable: ", using Default"

File: artemis-console-extension/artemis-extension/public/artemisconfig.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+    "jmx": {
+      "domain": "org.apache.activemq.artemis"
+    }
+}
```
Comment (lavocatt) [inline artemis-console-extension/artemis-extension/public/artemisconfig.json:5]: Is there some validation done in the pipelline at some point? I'm thinking about a jsonschema of some sort.

Comment (andytaylor) [reply artemis-console-extension/artemis-extension/public/artemisconfig.json:5]: > Is there some validation done in the pipelline at some point? I'm thinking about a jsonschema of some sort.

There isn't, tbh I took this code from hawtIO, it is how they do it with hawtconfig.json. I wouldnt know how to do whatyou suggest tbh.

Comment (lavocatt) [reply artemis-console-extension/artemis-extension/public/artemisconfig.json:5]: That's alright, that was mainly an out-of-curiosity question. No reason to diverge from hawtio in any case.

**Pull request merged**